File: c/array/majority-element.c
int majorityElement(int* nums, int numsSize) {
    int count;
    int ret;
    int i;

    count = 0;
    for(i = 0; i < numsSize; ++i){
        if(count == 0)
            ret = nums[i];
        if(ret == nums[i])
            count++;
        else
            count--;
    }

    return ret;
}

File: go/array/majority-element.go
func majorityElement(nums []int) int {
    var count, value int

    for i := range nums {
        if count == 0 {
            value = nums[i]
        }
        if value == nums[i] {
            count++
        }else{
            count--
        }
    }

    return value
}
